aiida_orca/parsers/__init__.py

```diff
@@ -24,25 +28,25 @@ def parse(self, **kwargs):
         If it would be an optimization run, the relaxed structure also will
         be stored under relaxed_structure key.
         """
-
         try:
             out_folder = self.retrieved
         except NotExistent:
             return self.exit_codes.ERROR_NO_RETRIEVED_FOLDER
 
-        fname_out = self.node.process_class._OUTPUT_FILE  #pylint: disable=protected-access
-        fname_relaxed = self.node.process_class._RELAX_COORDS_FILE  #pylint: disable=protected-access
-        # fname_traj = self.node.process_class._TRAJECTORY_FILE  #pylint: disable=protected-access
-        # fname_hessian = self.node.process_class._HESSIAN_FILE  #pylint: disable=protected-access
+        process_cls = self.node.process_class
+        fname_out = process_cls._OUTPUT_FILE  # pylint: disable=protected-access
+        fname_relaxed = process_cls._RELAX_COORDS_FILE  # pylint: disable=protected-access
 
-        if fname_out not in out_folder._repository.list_object_names():  #pylint: disable=protected-access
+        if fname_out not in out_folder.list_object_names():
             raise OutputParsingError('Orca output file not retrieved')
 
         try:
-            with self.retrieved.open(fname_out) as handler:
-                parsed_obj = ccread(handler.name)
+            with self.retrieved.open(fname_out, 'rb') as handle:
+                with tempfile.NamedTemporaryFile('w+b') as tmpfile:
```

Comment:
Is the NamedTemporaryFile needed here? In a PR where I implemented the AiiDA 2.0 support, I was able to simply pass the handler directly to `ccread`. 

https://github.com/pzarabadip/aiida-orca/pull/46/files#diff-442a1d8b7896fbb8c11128c00b321b263abe45a0c7134293d4e37f1d22e1a18eR41

Reply:
But did you test that code? Because if I run the test, it fails:
```
aiida_orca/parsers/cclib/ccio.py:48: in ccopen
    inputfile = logfileparser.openlogfile(source)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

filename = <_io.BufferedReader name='/tmp/tmpxsaxb48r/test_repo/repository/node/2b/77/4f79-56b5-46a1-b222-7882881e6b1c/path/aiida.out'>, object = None

    def openlogfile(filename, object=None):
        """Return a file object given a filename or if object specified decompresses it
        if needed and wrap it up.
    
        Given the filename or file object of a log file or a gzipped, zipped, or bzipped
        log file, this function returns a file-like object.
    
        Given a list of filenames, this function returns a FileInput object,
        which can be used for seamless iteration without concatenation.
        """
    
>       fileobject = FileWrapper(io.open(filename, 'r', errors='ignore'))
E       TypeError: expected str, bytes or os.PathLike object, not _io.BufferedReader
```
It is calling `io.open` which doesn't support file handles.

Reply:
Ah, in that PR I got rid of the `logfileparser.openlogfile()` call altogether as it is not needed, and that's I think how I made it work. Never mind then, I'll deal with this later after these changes are merged.